src/wallet/walletdb.cpp

```diff
@@ -311,6 +311,7 @@ class CWalletScanState {
     std::map<std::pair<uint256, CKeyID>, CKey> m_descriptor_keys;
     std::map<std::pair<uint256, CKeyID>, std::pair<CPubKey, std::vector<unsigned char>>> m_descriptor_crypt_keys;
     std::map<uint160, CHDChain> m_hd_chains;
+    bool tx_corrupt;
```

Comment:
```suggestion
    bool tx_corrupt{false};
```

To avoid potential use of uninitialized memory when the logic reading/writing `tx_corrupt` is changed over time.